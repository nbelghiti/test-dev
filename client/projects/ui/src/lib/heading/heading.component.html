<div
	class="c-heading {{ classes }}"
	[ngClass]="{'c-heading--inline': inline}"
>
	<ng-container [ngSwitch]="level">
		<h1 *ngSwitchCase="HeadingLevel.heading1">{{title}}</h1>
		<h2 *ngSwitchCase="HeadingLevel.heading2">{{title}}</h2>
		<h3 *ngSwitchCase="HeadingLevel.heading3">{{title}}</h3>
	</ng-container>

	<div class="c-heading__meta">
		<i
			*ngIf="loading; else withContent"
			class="fas fa-{{IconName.spinner}} fa-spin c-heading__loader"
		>
		</i>

		<ng-template #withContent>
			<ng-content></ng-content>
		</ng-template>
	</div>
</div>
