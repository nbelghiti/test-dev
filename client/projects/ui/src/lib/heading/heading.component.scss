// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Heading
// =============================================================================
.c-heading {
	display: inline-flex;
	flex-direction: column;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-heading__loader {
	color: $primary-color;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-heading--inline {
	flex-direction: row;
	align-items: baseline;

	.c-heading__meta {
		margin-top: 0;
		margin-left: 1rem;
	}
}
