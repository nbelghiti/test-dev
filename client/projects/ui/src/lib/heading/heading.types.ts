export enum HeadingLevel {
	heading1 = '1',
	heading2 = '2',
	heading3 = '3',
}
