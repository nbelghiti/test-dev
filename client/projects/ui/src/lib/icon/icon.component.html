<i class="c-icon__glyph {{ size && 'c-icon__glyph--' + size }} {{ type && 'c-icon__glyph--' + type }} fas fa-{{ name }}"></i>
