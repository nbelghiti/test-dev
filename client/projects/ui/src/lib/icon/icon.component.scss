// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Icon
// =============================================================================
:host {
	display: inline-flex;
	justify-content: center;
	align-items: center;

	font-size: inherit;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-icon__glyph {
	line-height: inherit;

	cursor: pointer;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-icon__glyph--large {
	font-size: 2rem;
}

.c-icon__glyph--small {
	font-size: 1.4rem;
}

.c-icon__glyph--tiny {
	font-size: 1.2rem;
}

.c-icon__glyph--success {
	color: $ui-color-success-dark;
}

.c-icon__glyph--error {
	color: $ui-color-error-dark;
}

.c-icon__glyph--warning {
	color: $ui-color-warning-dark;
}

.c-icon__glyph--info {
	color: $ui-color-info-dark;
}
