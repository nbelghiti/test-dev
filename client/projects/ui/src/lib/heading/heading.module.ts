import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';

import { HeadingComponent } from './heading.component';

@NgModule({
	imports: [
		CommonModule,
	],
	declarations: [
		HeadingComponent,
	],
	exports: [
		HeadingComponent,
	],
})
export class HeadingModule {}
