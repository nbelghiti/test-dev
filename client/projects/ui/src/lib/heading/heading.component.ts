import { Component, Input } from '@angular/core';

import { IconName } from '../icon/icon.types';

import { HeadingLevel } from './heading.types';

@Component({
	selector: 'ui-heading',
	templateUrl: './heading.component.html',
	styleUrls: ['./heading.component.scss']
})

export class HeadingComponent {
	@Input() public title: string;
	@Input() public classes: string = '';
	@Input() public inline: boolean = false;
	@Input() public loading: boolean = false;
	@Input() public level: HeadingLevel = HeadingLevel.heading1;

	public HeadingLevel: typeof HeadingLevel = HeadingLevel;
	public IconName: typeof IconName = IconName;
}
