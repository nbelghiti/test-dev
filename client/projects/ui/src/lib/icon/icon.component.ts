import { Component, HostBinding, Input } from '@angular/core';

import { IconName, IconSize, IconType } from './icon.types';

@Component({
	selector: 'ui-icon',
	templateUrl: './icon.component.html',
	styleUrls: ['./icon.component.scss']
})

export class IconComponent {
	@Input() public name: IconName;
	@Input() public type: IconType = null;
	@Input() public size: IconSize = null;

	@HostBinding('class.c-icon') public className = true;
}
