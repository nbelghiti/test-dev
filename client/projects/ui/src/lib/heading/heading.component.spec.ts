import { ComponentFixture, TestBed, async } from '@angular/core/testing';

import { IconModule } from '../icon/icon.module';

import { HeadingComponent } from './heading.component';

describe('HeadingComponent', () => {
	let component: HeadingComponent;
	let fixture: ComponentFixture<HeadingComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			imports: [
				IconModule,
			],
			declarations: [
				HeadingComponent
			]
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(HeadingComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});
