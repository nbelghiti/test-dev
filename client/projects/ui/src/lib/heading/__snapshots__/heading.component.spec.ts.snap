// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadingComponent Should render correctly 1`] = `
<ui-heading
  HeadingLevel={[Function Object]}
  IconName={[Function Object]}
  classes=""
  inline="false"
  level={[Function String]}
  loading="false"
>
  <div
    class="c-heading"
    ng-reflect-klass="c-heading "
    ng-reflect-ng-class="[object Object]"
  >
    
    
    <h1>
      
    </h1>
    
    
    <div
      class="c-heading__meta"
    >
      
      
      
    </div>
  </div>
</ui-heading>
`;
